src/ui/preview/Preview.vue

```diff
@@ -62,19 +68,29 @@ export default {
             viewKey: undefined,
             views: [],
             currentView: {},
-            actionCollection: undefined
+            actionCollection: undefined,
+            existingViewIndex: 0
         };
     },
     mounted() {
-        this.views = this.openmct.objectViews.get(this.domainObject, this.objectPath).map((view) => {
+        this.views = this.openmct.objectViews.get(this.domainObject, this.objectPath).map((view, index) => {
```

Comment:
Two issues here that are nothing to do with your changes but which we should fix anyway.

This function returns view _providers_ not views. I think if we were to rename `this.views` to `this.viewProviders` it will help make some of the subsequent code easier to understand (at least for me).

Also, why is this a `map` function? It doesn't map any values, it should be a `forEach`.

Reply:
Gotcha!